test/e2e/specs/editor/various/format-library/math.spec.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'Format Library - Math', () => {
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
+	test( 'should insert math format with LaTeX', async ( {
+		editor,
+		page,
+		pageUtils,
+	} ) => {
+		await editor.canvas
+			.getByRole( 'button', { name: 'Add default block' } )
+			.click();
+
+		await page.keyboard.type( 'equation: ' );
+
+		await editor.clickBlockToolbarButton( 'More' );
+		await page.getByRole( 'menuitem', { name: 'Math' } ).click();
+
+		expect( await editor.getBlocks() ).toMatchObject( [
```

Comment:
```suggestion
		await expect.poll( editor.getBlocks ).toMatchObject( [
```

It's better to use `await expect.poll` and async assertions, which benefit from auto-retries. The same applies to similar assertions in this file.

Reply:
Created PR to fix this in the codebase - https://github.com/WordPress/gutenberg/pull/73153.

Reply:
I never got this. Why is it not ok to expect this to be correct immediately?

Reply:
It's more like a safeguard or best practice. Most of the time, it's correct, and the test passes immediately, with no additional overhead. However, when it's not, the auto-retries are helpful to ensure that failure is actually related to not matching the object.

Reply:
Not a blocker here. We can continue talking in the new PR.